### PR TITLE
[MRG] Update documentation of `perimeter` and add input validation

### LINKED ISSUE #3550: perimeter documentation and prototype function
## Description
skimage.measure.perimeter lacks a relevant documentation and description.

The neighborhood configuration should precise that it is used for computing the contour and is different from the connectivity of the contour itself. 

Moreover, the general terminology (1-, 2-, 3-, ..., ndim-) should be adopted in the prototype of the function.

### PR DESCRIPTION
**Pull request by Henley13**
## Description
Update documentation of perimeter, add an example and raise an error if the input image is in 3D. 

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [x] Gallery example in `./doc/examples` (new features only)
- [x] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [x] Unit tests

## References
Closes #3550 

## For reviewers
- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/measure/_regionprops.py [hunk 1]
```diff
@@ -601,7 +603,25 @@ def perimeter(image, neighbourhood=4):
     .. [1] K. Benkrid, D. Crookes. Design and FPGA Implementation of
            a Perimeter Estimator. The Queen's University of Belfast.
            http://www.cs.qub.ac.uk/~d.crookes/webpubs/papers/perimeter.doc
+
+    Examples
+    --------
+    >>> from skimage import data, util
+    >>> from skimage.measure import label
+
+    # coins image (binary)
```
Comment (soupault) [inline skimage/measure/_regionprops.py]: I'm not sure this is correct syntax. Have you tried to check by building the documentation locally?

File: skimage/measure/_regionprops.py [hunk 2]
```diff
@@ -601,7 +603,25 @@ def perimeter(image, neighbourhood=4):
...
+    >>> img_coins = util.img_as_ubyte(data.coins()) > 110
+
+    # total perimeter of all objects in the image
+    >>> perimeter(img_coins, neighbourhood=4)
+    7796.8679964360044
+    >>> perimeter(img_coins, neighbourhood=8)
+    8806.2680733252855
+
     """
+    if image.ndim > 2:
```
Comment (soupault) [inline skimage/measure/_regionprops.py]:
```suggestion
    if image.ndim != 2:
```

File: skimage/measure/_regionprops.py [hunk 3]
```diff
@@ -601,7 +603,25 @@ def perimeter(image, neighbourhood=4):
...
+        raise NotImplementedError('perimeter does not support 3D images')
```
Comment (soupault) [inline skimage/measure/_regionprops.py]:
```suggestion
        raise NotImplementedError('`perimeter` supports 2D images only')
```

File: skimage/measure/_regionprops.py [hunk 4]
```diff
@@ -586,10 +586,12 @@ def perimeter(image, neighbourhood=4):
 
     Parameters
     ----------
-    image : array
-        Binary image.
+    image : (N, M) ndarray
+        2D binary image.
     neighbourhood : 4 or 8, optional
-        Neighborhood connectivity for border pixel determination.
+        Neighborhood connectivity for border pixel determination. It's used to
```
Comment (emmanuelle) [inline skimage/measure/_regionprops.py]: It's --> It is (sorry we are a bit formal, like in papers :-))

File: skimage/measure/_regionprops.py (hunk 1 above)
Comment (Henley13) [reply skimage/measure/_regionprops.py]: I removed the blank line and add some `>>>`.

I get two warnings from the build : 
- WARNING: the sphinx_copybutton extension does not declare if it is safe for parallel reading, assuming it isn't - please ask the extension author to check and make it explicit
- WARNING: doing serial read

Except that, it seems ok.

File: skimage/measure/_regionprops.py [hunk 5]
```diff
@@ -601,7 +603,23 @@ def perimeter(image, neighbourhood=4):
     .. [1] K. Benkrid, D. Crookes. Design and FPGA Implementation of
            a Perimeter Estimator. The Queen's University of Belfast.
            http://www.cs.qub.ac.uk/~d.crookes/webpubs/papers/perimeter.doc
+
+    Examples
+    --------
+    >>> from skimage import data, util
+    >>> from skimage.measure import label
+    >>> # coins image (binary)
+    >>> img_coins = util.img_as_ubyte(data.coins()) > 110
```
Comment (sciunto) [inline skimage/measure/_regionprops.py]:
```suggestion
     >>> img_coins = data.coins() > 110
```

I don't think the conversion is necessary here.

Comment (Henley13) [reply skimage/measure/_regionprops.py]: Yes it works !

**Pull request merged**